src/worker.cpp

```diff
@@ -1185,4 +1184,281 @@ void Worker::collectFrusFromJson()
+    }
+
+    if (!l_primaryHwPath.empty() && !l_secondaryHwPath.empty() &&
+        l_parsedPrimaryVpdMap.index() != l_parsedSecVpdMap.index())
+    {
+        logging::logMessage(
+            "Primary and secondary VPD types are not matching.");
+        return;
+    }
+
+    // Implement backup and restore for ipz type VPD
+    if (std::holds_alternative<types::IPZVpdMap>(l_parsedPrimaryVpdMap) ||
+        std::holds_alternative<types::IPZVpdMap>(l_parsedSecVpdMap))
+    {
+        if (const auto& l_backupInfoMap =
+                std::get_if<types::IpzBackupMap>(&l_backupMap))
+        {
+            auto l_primaryVpdMap = types::IPZVpdMap{};
+            if (!l_primaryHwPath.empty())
+            {
+                const auto l_ipzVpd =
+                    std::get_if<types::IPZVpdMap>(&l_parsedPrimaryVpdMap);
+                l_primaryVpdMap = *l_ipzVpd;
+            }
+
+            auto l_secondaryVpdMap = types::IPZVpdMap{};
+            if (!l_secondaryHwPath.empty())
+            {
+                const auto l_ipzVpd =
+                    std::get_if<types::IPZVpdMap>(&l_parsedSecVpdMap);
+                l_secondaryVpdMap = *l_ipzVpd;
+            }
+
+            if (std::holds_alternative<std::monostate>(io_parsedVpdMap))
+            {
+                auto l_parsedVpdMap = types::IPZVpdMap{};
+                backupAndRestoreIpzVpd(l_parsedVpdMap, *l_backupInfoMap,
+                                       l_primaryVpdMap, l_secondaryVpdMap);
+            }
+            else if (auto l_parsedVpdMap =
+                         std::get_if<types::IPZVpdMap>(&io_parsedVpdMap))
+            {
+                backupAndRestoreIpzVpd(*l_parsedVpdMap, *l_backupInfoMap,
+                                       l_primaryVpdMap, l_secondaryVpdMap);
+            }
+            {
+                logging::logMessage("Parsed VPD is not of IPZ type.");
+                return;
+            }
+        }
+        else
+        {
+            logging::logMessage("The backup map is not IPZ type.");
+            return;
+        }
+    }
+    else if (std::holds_alternative<types::KeywordVpdMap>(
+                 l_parsedPrimaryVpdMap) ||
+             std::holds_alternative<types::KeywordVpdMap>(l_parsedSecVpdMap))
+    {
+        // To Do, handle implementation for Keyword type VPD
+    }
+}
+
+void Worker::backupAndRestoreIpzVpd(types::IPZVpdMap& io_parsedVpdMap,
+                                    const types::IpzBackupMap& i_backupInfoMap,
+                                    const types::IPZVpdMap& i_primaryVpdMap,
+                                    const types::IPZVpdMap& i_secondaryVpdMap)
+{
+    const std::string& l_primaryHwPath = getPrimaryHwPath();
+    const std::string& l_secondaryHwPath = getSecondaryHwPath();
+
+    for (const auto& l_recordKwInfo : i_backupInfoMap)
+    {
+        const std::string& l_recordName = l_recordKwInfo.first;
+        if (!io_parsedVpdMap.empty() &&
```

Comment:
Check for empty parsed VPD map in the beginning. No need to check in each iteration of the loop.

Reply:
As io_parsedVpdMap is not mandatory to be given(no empty).
if its not an empty map, checking for record present otherwise will continue with other records.